braidpool-primitives/src/bead/mod.rs

```diff
@@ -1,60 +1,53 @@
+#![allow(unused)]
 // Standard Imports
+use ::serde::{Deserialize, Serialize};
 use std::collections::HashSet;
+use std::net::SocketAddr;
 
 // Bitcoin primitives
+use bitcoin::Address;
 use bitcoin::absolute::Time;
+use bitcoin::ecdsa::Signature;
+use bitcoin::p2p::Address as P2P_Address;
+use bitcoin::secp256k1::PublicKey;
 use bitcoin::transaction::TransactionExt;
-use bitcoin::{BlockHeader, CompactTarget, Transaction};
-
+use bitcoin::{BlockHeader, Transaction};
 // Custom Imports
 use crate::utils::BeadHash;
-use crate::utils::bitcoin::MerklePathProof;
-
-// Type Aliases
-type TransactionWithMerklePath = (Transaction, MerklePathProof);
-
-pub struct Bead {
-    pub block_header: BlockHeader,
-    pub bead_hash: BeadHash,
-    pub coinbase_transaction: TransactionWithMerklePath,
-    pub payout_update_transaction: TransactionWithMerklePath,
+#[derive(Clone, Debug, Serialize)]
 
+pub struct CommittedMetadata {
     // Committed Braidpool Metadata,
-    pub lesser_difficulty_target: CompactTarget,
     pub parents: HashSet<(BeadHash, Time)>,
```

Comment:
When we serialize a HashSet, can the order be depended upon?

I'm thinking maybe we move the timestamps into UncommittedMetadata -- my question is, if the serialization of `parents` is effectively a list, can we put a timestamp list in `UncommittedMetadata` such that the list of timestamps is in the same order as this `parents` serialization?